q2_composition/tests/test_ancombc.py

```diff
@@ -53,7 +53,97 @@ def test_ids_in_table_not_in_md(self):
             ancombc(table=self.table, metadata=self.missing_md,
                     formula='bodysite')
 
-    # confirm level ordering behavior
+    # confirm output columns based on formula inputs and ref levels
+    def test_output_cols_single_formula_no_ref_level(self):
+        exp_col_list = ['id', '(Intercept)', 'bodysiteleft palm',
+                        'bodysiteright palm', 'bodysitetongue']
+        obs_col_list = []
+
+        dataloaf = ancombc(table=self.table, metadata=self.md,
+                           formula='bodysite')
+
+        slices = dataloaf.data_slices.iter_views(pd.DataFrame)
+        for _, slice in slices:
+            for col in slice.columns:
+                obs_col_list.append(col)
+                self.assertNotIn('bodysitegut', col)
+
+        self.assertEqual(set(exp_col_list), set(obs_col_list))
+
+    def test_output_cols_single_formula_single_ref_level(self):
+        exp_col_list = ['id', '(Intercept)', 'bodysiteleft palm',
+                        'bodysiteright palm', 'bodysitegut']
+        obs_col_list = []
+        dataloaf = ancombc(table=self.table, metadata=self.md,
+                           formula='bodysite',
+                           reference_levels=['bodysite::tongue'])
+
+        slices = dataloaf.data_slices.iter_views(pd.DataFrame)
+        for _, slice in slices:
+            for col in slice.columns:
+                obs_col_list.append(col)
+                self.assertNotIn('bodysitetongue', col)
+
+        self.assertEqual(set(exp_col_list), set(obs_col_list))
```

Comment:
Not sure if this is the best way to handle, but this was the quickest way to deal w/the observed columns since they are repeated for each slice in the dataloaf & check that ALL of the expected columns were present & nothing else.

Reply:
Seems like a reasonable approach to me.

Reply:
Unresolved because I edited the comment while looking things over again